tuist/tuist
# Fix issue with MacCatalyst archiving not being included in build cache when caching `XCFramework`

Resolves https://github.com/tuist/tuist/issues/4900

### Short description 📝

Update `CacheXCFrameworkBuilder` to generate MacCatalyst archives for `XCFramework` based on their support status when creating a cache.

### How to test the changes locally 🧐

Download the sample project through this [link](https://github.com/devyhan/TuistMacCatalystExample/archive/refs/heads/main.zip) and enter the following commands according to the CLI. Afterward, MacCatalyst should be added to the BuildCache as shown in the image below, and the cached sample project should build the MacCatalyst target.

1. tuist fetch
2. tuist cache warm --xcframeworks
3. tuist generate --xcframeworks

*Check the path  `~/.tuist/Cache/BuildCache/`*

![image](https://user-images.githubusercontent.com/45344633/225569022-b1596b3b-f4a3-4193-ba91-9e2d156514ab.png)

### Contributor checklist ✅


### Reviewer checklist ✅
